### ISSUE TITLE
Meta Bind is better

### ISSUE THREAD

**Issue by shabegom** [shabegom/buttons#208]
Y'all should use this new Meta Bind plugin. It has slick Buttons!

https://github.com/mProjectsCode/obsidian-meta-bind-plugin

**Comment by SackaryBorks** [comment]
Does this mean that this project is no longer going to be supported?

Recently adopted this and I enjoy how easy it was to understand and start using in my work notes. While they may boast more features and include buttons, after working with theirs I do not want to switch.

**Comment by thaytuh** [comment]
I agree with Sackary, just the fact that Meta Bind doesn't support dynamic button naming conventions such as automatically pulling the date makes it less enjoyable to use. But at the same time I've been having issues with Buttons not working

**Comment by MRobinson-dev** [comment]
I can't get Meta Bind to work to create a new note with templater

**Comment by sensiblearts** [comment]
I'm new to Obsidian and to Buttons.

Right now, I'd really like to put 2 actions in a button, e.g.,

```button
name MyButton
type command
action Search: Search in all files
type link
action file://C:/Program Files/Scrivener3/Scrivener.exe
```

Is that fairly doable with Buttons? (If so, I could learn the plugin code and do it if it.)

Or, can Meta Bind already do that?

Also, what would be cool is to pass search parameters to `action Search`...

Thanks,
David Alm

**Comment by thaytuh** [comment]
I'm still able to use Buttons with Templater, but I've had to to do some workarounds. I notice the button doesn't work to properly create the new note unless I'm in Live preview mode, for some reason. As for the auto renaming, that should entirely be a templater function, separate from Buttons or Meta Bind as far as I know.
I use this command to rename my files: `<% tp.file.rename(tp.date.now("YYYY-MM-DD") + " NameOfNoteHere") %>`

**Comment by thaytuh** [comment]
Oh yeah, I also had an issue where my buttons would disappear/stop working after I opened and synced Obsidian on my phone. Never really found a fix, it just stopped happening. Also, I just have that string of templater code inside the template itself, my button is just a normal button that creates a note from that template. The code being inside the template is what triggers it, at least in my experience. Let me know how Meta Bind works, as I've been considering the switch but have just been getting by with Buttons so far.

**Comment by elipsonknight14** [comment]
@thaytuh 

I am far from a coder and I barely understand how to get this stuff working. Once I get it going once, its kinda shocking and then a set it and forget it kinda thing.  This is what I was using to create a file and auto name that file. Are you saying that the file renaming can be added to the template and that will solve my problems?

```button
name New Visit
type note(247 - Visit <% tp.date.now("YYYY-MM-DD ddd") %>)) template
action 247 Visit Template
templater true
color purple
```

**Comment by thaytuh** [comment]
This is how I have mine set up:
`name New Morning Manifesto
type note(Untitled) template
action Morning Manifesto
color red
folder 03 - daily/Morning Manifesto
prompt false`
Try changing the part of your button that says `type note(247 - Visit <% tp.date.now("YYYY-MM-DD ddd") %>)) template` to just say `type note(Untitled) template` and also remove the part that says `templater true`. I'm not 100% sure this will work so I would suggest copying the entire text block for the button and pasting it after, essentially creating a second button to mess with.
Then, inside the base template note, add this line somewhere on the page (I put mine on the bottom or in between headings as the text disappears when the note is created but it leaves an empty line):
`<% tp.file.rename("247 - Visit " + tp.date.now("YYYY-MM-DD")) %>`
I can't guarantee this will work, I just know this is what works with my current setup. But let me know if you need anything else, I'm here to help :)

**Comment by elipsonknight14** [comment]
@thaytuh 

OMG! That worked for Renaming it correctly! Thank you!

I see in you example this line
"folder 03 - daily/Morning Manifesto"

Is that telling the system what folder to create the note into? I'd like to do that too because now that the Renaming is functioning, it seems to have made my tp.file.move script not work. I can get it to rename OR move but not both.

**Comment by thaytuh** [comment]
I'm so glad I could help! My Kofi is https://ko-fi.com/thaytuh if you'd like to help out. I'm always available for questions for whatever, but I still have a lot to learn. :)
